python-avd/pyavd/_eos_designs/shared_utils/overlay.py

```diff
@@ -66,7 +66,7 @@ def get_rd_admin_subfield_value(self: SharedUtilsProtocol, admin_subfield: str,
             return self.vtep_ip
 
         if admin_subfield == "bgp_as":
-            return cast("str", self.bgp_as)
+            return cast("str", self.get_asn(self.bgp_as))
```

Comment:
does it work with dot notation for this?

Reply:
```
+router bgp 555.65003
+   bgp asn notation asdot
    router-id 192.168.255.3
    update wait-install
    no bgp default ipv4-unicast
@@ -127,17 +128,17 @@ router bgp 65003
    redistribute connected route-map RM-CONN-2-BGP
    !
    vlan 1
-      rd 65003:10001
+      rd 555.65003:10001
       route-target both 15:10001
       redistribute learned
    !
    vlan 2
-      rd 65003:10002
+      rd 555.65003:10002
       route-target both 15:10002
       redistribute learned
```

Reply:
but maybe we should always translate back to asplain notation to avoid diffs

Reply:
so lets make sure we have a conversion for this that will always convert back to asplain as it is what is rendered by EOS thanks!

Reply:
thanks for catching this up though I am not sure we are better off with this behavior

it means if a user enters an integer for `admin_subfield` we also need to convert it to the asdot value so that requires some extra work as well

```
inet-cloud(config-router-bgp)#vlan 2
inet-cloud(config-macvrf-2)#rd 123456123:45
inet-cloud(config-macvrf-2)#vlan 2
inet-cloud(config-macvrf-2)#show active
router bgp 555.65003
   vlan 2
      rd 1883.51835:45
```

@Vibhu-gslab please add some tests covering the various admin_subfield in `overlay_rd_type` that actually use BGP AS numbers > 65536 and check the various outputs to make sure we address things properly